.buildkite/hooks/pre-command

```diff
@@ -14,8 +14,15 @@ if [[ "$BUILDKITE_COMMAND" =~ .*"upload".* ]]; then
   exit 0
 fi
 
+# Run always
+source .buildkite/hooks/prepare-common.sh
```

Comment:
Since I changed when to install things, I ran the snapshot workflow manually:
- https://buildkite.com/elastic/apm-agent-java-snapshot/builds/397

In addition, I'm also validating these changes in the Buildkite pipeline:
- https://buildkite.com/elastic/apm-agent-java-opentelemetry-benchmark/builds/46

Reply:
Was the snapshot actually run on this branch?
According to the logs it still ran on Java 17 and I can't see how this would be the case with this PR:
```
Java version: 17.0.2, vendor: Oracle Corporation, runtime: /usr/java/openjdk17
```

We will need to stick to Java 17 for building the agent for Java 7 support.

Reply:
This change won't affect snapshot/releases Buildkite pipelines but the opentelemetry benchmark only:
- https://buildkite.com/elastic/apm-agent-java-opentelemetry-benchmark/builds?branch=fix%2Fjava-21-otel

<img width="579" alt="image" src="https://github.com/elastic/apm-agent-java/assets/2871786/b6b9b9e8-0522-40e3-9ad5-e143b755c814">

the it failed with

```

> Task :smoke-tests-otel-starter:spring-smoke-testing:bootJar FAILED
--
  |  
  | FAILURE: Build failed with an exception.
  |  
  | * What went wrong:
  | Execution failed for task ':smoke-tests-otel-starter:spring-smoke-testing:bootJar'.
  | > Error while evaluating property 'mainClass' of task ':smoke-tests-otel-starter:spring-smoke-testing:bootJar'.
  | > Failed to calculate the value of task ':smoke-tests-otel-starter:spring-smoke-testing:bootJar' property 'mainClass'.
  | > Main class name has not been configured and it could not be resolved


```

when running https://github.com/elastic/apm-agent-java/blob/ea685113257279c820dc4b97d288486fb7cbe76d/.buildkite/scripts/opentelemetry-benchmark.sh#L39

Reply:
Ok then I don't see a reason why this shouldn't work, if this only affects the benchmark. @jackshirazi ?

Reply:
yes, let's go ahead